### PR TITLE
✨ amp-next-page: Allow hiding of arbitrary elements in child pages

### PR DESCRIPTION
**Pull request by peterjosling**
Primary use case is hiding repeated page headers/footers.

- Reads from hideSelectors key in config
- Blacklists i-amphtml-* selectors, like the validator
- Runs querySelectorAll() for all the selectors when the shadow document is attached. Generating and injecting CSS would presumably be more efficient but would need more validation/sanitisation on the selectors. AFAIK passing user input directly to `querySelectorAll` should be safe.

### CODE REVIEW

**Comment by peterjosling** [comment]
@aghassemi

File: extensions/amp-next-page/0.1/config.js [hunk 1]
```diff
@@ -51,13 +50,35 @@ export function assertConfig(config, origin, sourceOrigin) {
...
 }
 
+const BANNED_SELECTOR_PATTERNS = [
+  /(^|\W)i-amphtml-/,
+];
+
+function assertSelectors(selectors) {
+  selectors.forEach(selector => {
+    BANNED_SELECTOR_PATTERNS.forEach(pattern => {
+      user().assert(typeof selector === 'string',
```
Comment (aghassemi) [inline extensions/amp-next-page/0.1/config.js]: `user().assertString`

Comment (peterjosling) [reply extensions/amp-next-page/0.1/config.js]: Done, and updated other usages.

File: extensions/amp-next-page/0.1/test/test-amp-next-page.js [hunk 2]
```diff
@@ -124,4 +139,60 @@ env => {
...
+    sandbox.stub(viewport, 'getClientRectAsync').callsFake(() => {
+      // 1x viewport away
+      return Promise.resolve(
+          layoutRectLtwh(0, 0, sizes.width, sizes.height * 2));
+    });
+
+    win.dispatchEvent(new Event('scroll'));
+    yield macroTask();
+
+    const shadowDoc = attachShadowDocSpy.firstCall.returnValue.ampdoc;
```
Comment (rsimha) [inline extensions/amp-next-page/0.1/test/test-amp-next-page.js]: @aghassemi @peterjosling This test is causing widespread unit test failures on `master` today: https://travis-ci.org/ampproject/amphtml/jobs/377848593#L2008

Can you fix this test, or skip if the fix isn't obvious?

**Pull request merged**